chain-signatures/node/src/mesh/connection.rs

```diff
@@ -68,24 +69,47 @@ impl Pool {
 
         let connections = self.connections.read().await;
 
-        let mut status = self.status.write().await;
         let mut participants = Participants::default();
+
+        // FuturesUnordered for parallel execution
+        let mut tasks = FuturesUnordered::new();
```

Comment:
There are issues with using `FuturesUnordered` outlined in github.com/rust-lang/futures-rs/issues/2387. Better to use `tokio::task::JoinSet` for this